src/main/java/com/robertx22/age_of_exile/gui/texts/StatCategory.java

```diff
@@ -0,0 +1,56 @@
+package com.robertx22.age_of_exile.gui.texts;
+
+import com.google.common.collect.Sets;
+import com.robertx22.age_of_exile.aoe_data.database.ailments.Ailment;
+import com.robertx22.age_of_exile.database.data.stats.types.defense.Armor;
+import com.robertx22.age_of_exile.database.data.stats.types.defense.DodgeRating;
+import com.robertx22.age_of_exile.database.data.stats.types.offense.WeaponDamage;
+import com.robertx22.age_of_exile.database.data.stats.types.resources.RegeneratePercentStat;
+import com.robertx22.age_of_exile.database.data.stats.types.resources.energy.Energy;
+import com.robertx22.age_of_exile.database.data.stats.types.resources.energy.EnergyRegen;
+import com.robertx22.age_of_exile.database.data.stats.types.resources.health.Health;
+import com.robertx22.age_of_exile.database.data.stats.types.resources.health.HealthRegen;
+import com.robertx22.age_of_exile.database.data.stats.types.resources.magic_shield.MagicShield;
+import com.robertx22.age_of_exile.database.data.stats.types.resources.magic_shield.MagicShieldRegen;
+import com.robertx22.age_of_exile.database.data.stats.types.resources.mana.Mana;
+import com.robertx22.age_of_exile.database.data.stats.types.resources.mana.ManaRegen;
+import com.robertx22.age_of_exile.database.registry.ExileDB;
+import com.robertx22.age_of_exile.saveclasses.item_classes.tooltips.TooltipStatInfo;
+import lombok.AllArgsConstructor;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+//use to sort the stat when showing MergeStats.
+@AllArgsConstructor
+public enum StatCategory {
+    SDI(x -> Sets.newHashSet("all_attributes", "strength", "dexterity", "intelligence").contains(x)),
+    CORE(x -> Sets.newHashSet(WeaponDamage.GUID, Health.GUID, MagicShield.GUID, Mana.GUID, Energy.GUID, Armor.GUID, DodgeRating.GUID, "critical_hit", "critical_damage").contains(x)),
+    REGEN(x -> Sets.newHashSet(HealthRegen.getInstance().GUID(), MagicShieldRegen.getInstance().GUID(), ManaRegen.getInstance().GUID(), EnergyRegen.getInstance().GUID(), RegeneratePercentStat.HEALTH.GUID(), RegeneratePercentStat.MAGIC_SHIELD.GUID(), RegeneratePercentStat.MANA.GUID(), RegeneratePercentStat.ENERGY.GUID()).contains(x)),
+    RESISTS(x -> x.contains("_resist")),
```

Comment:
Lots of hard coded strings that will break if i ever rename stat id

Reply:
left ELE_DAMAGE unchanged cuz I don't wanna do that reflect thing again like in DEFENSE, it's horrible.